drgn/helpers/linux/net.py

```diff
@@ -37,6 +39,27 @@ def for_each_net(prog: Program) -> Iterator[Object]:
         yield net
 
 
+_CLONE_NEWNET = 0x40000000
+
+
+def get_net_ns_by_file(file: Object) -> Object:
+    """
+    Get the network namespace corresponding to the given network namespace (NSFS)
+    pseudo-file, e.g. '/proc/$PID/ns/net' or '/var/run/netns/$NAME'.
+
+    :param file: ``struct file *`` network namespace pseudo-file.
+    :return: ``struct net *`` (``NULL`` if invalid file)
+    """
+    if file.f_op != file.prog_["ns_file_operations"].address_of_():
+        return NULL(file.prog_, "struct net *")
```

Comment:
I think we should raise an exception instead. Maybe:

```suggestion
        raise ValueError("not a net namespace inode")
```